### PR TITLE
ab facs: validate on publish

### LINKED ISSUE #668: AB FACS: validate experiments before publishing
As an instructor, I can't publish a FACS experiment without a complete set of histograms.

When an instructor attempts to preview or publish a FACS experiment with an incomplete set of histograms, we should return an error message (similar to Western Blot)

### PR DESCRIPTION
**Pull request by annagav**
#### What are the relevant tickets?

Fix #668
#### What's this PR do?

When the user clicks 'publish' in the dashboard, the validation checks if every 'sample_prep' object in FACS has histogram data.

### CODE REVIEW

File: instructor/common.py [hunk 1]
```diff
@@ -1738,24 +1738,39 @@ def assignment_complete(request):
 
 
 def is_assignment_complete(assignment):
-    can_preview = False
     if models.StrainTreatment.objects.filter(
         assignment=assignment,
         enabled=True
     ).exists():
-        if assignment.has_wb and models.WesternBlotBands.objects.filter(
+        if assignment.has_fc or assignment.has_wb:
```
Comment (noisecapella) [inline instructor/common.py:1840]: What about microscopy?

File: instructor/common.py [hunk 2]
```diff
@@ -1738,24 +1738,39 @@ def assignment_complete(request):
...
+def is_facs_complete(assignment):
+    if assignment.has_fc:
+        if models.FlowCytometryHistogramMapping.objects.filter(
+            sample_prep__assignment=assignment
         ).exists():
-            can_preview = True
-        if (
-            assignment.has_fc and
-            models.FlowCytometryHistogramMapping.objects.filter(
+            mappings = models.FlowCytometryHistogramMapping.objects.filter(
```
Comment (noisecapella) [inline instructor/common.py]: Can you move this outside the if statement, then use `mappings.exists()` above?

File: instructor/common.py (hunk 1 above)
Comment (annagav) [inline instructor/common.py:1840]: Microscopy is not complete yet. Cannot publish Microscopy technique yet.

**Comment by noisecapella** [comment]
When I clicked 'preview' on the assignment I got this error:

screenshot from 2016-08-03 14-17-26

**Comment by annagav** [comment]
This is a migration issue. I rebased this branch, try again.

**Comment by noisecapella** [comment]
Code looks good, testing functionality now...

**Comment by annagav** [comment]
@noisecapella Should be fixed now.

**Comment by noisecapella** [comment]
'Add histogram' works now but I'm getting this exception when I click 'preview': 
screenshot from 2016-08-05 15-07-23

**Comment by noisecapella** [comment]
Nevermind, I forgot to migrate

**Comment by noisecapella** [comment]
:+1:

**Pull request merged**